Panel map

Gitek1 opened the pull request:
When we click on the map, it will resize over the content.

Gitek1 committed:
-added fullscreen API, which lets you bring panel map into fullscreen.
-added functionality on that big google map panel: navigations control,  add custom control button on google map for switch on/off weather layer)
-added css style for fullscreen mode
-added event's pop up for nodes......

Gitek1 committed:
- set option draggable on true

Gitek1 committed:
-added Google Earth integration on Google Maps Api V3

Gitek1 committed:
Added OpenStreet map as a base layer on Google Map API V3.

Gitek1 committed:
Fixed one click on map for fullscreen mode.
Changed css style with java script for non-default buttons on google map(they look like default google maps buttons now).

Gitek1 committed:
- fix interactive big panel map
- add button exit to that panel

Gitek1 committed:
Do slide down animation over the all content on the page for interactive map.

Gitek1 committed:
When we click on the map, it will resize over the content.

Gitek1 committed:
Improved CSS, so when we resize map over the content, the page in background is not visible.
Improved animation of resize in/out map.

Gitek1 committed:
Add close button to the right top corner of the advanced map.
Moved some existing code for opening advanced map to function open_advanced_map().

Gitek1 committed:
Fixed CSS style.
Defined local variable myOptions and nodeLocation.

Gitek1 committed:
Replaced previous default button for resizing map with new image button, which is positioned on the left edge of the map.

Gitek1 committed:
Added image extend button to static directory.

Gitek1 committed:
Defined google maps controls(draggable, scaleControl,panControl, navigationControl..), which become visible when we resize the map.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.css:
```diff
@@ -1,4 +1,92 @@
-#map_canvas {
-    width: 450px;
-    height: 450px;
+#basic-map{
```
Space before `{`.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.css:
```diff
@@ -1,4 +1,92 @@
+    height: 250px;
+    margin-bottom:50px;
+    margin-right:20px;
+    position:relative;
```
Space after `:`.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.css:
```diff
@@ -1,4 +1,92 @@
+    opacity: 1;
+    background-color:#333;
+    border:1px solid #666;
+    /* CSS3 styling for latest browsers */
```
No need for this comment.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.css:
```diff
@@ -1,4 +1,92 @@
+    border: none;
+}
+#advanced-map .close {
+    background-image:url(http://jquerytools.org/media/img/overlay/close.png);
```
Download this locally and add it to static files. We do not want to depend on third-party servers.

mitar commented:
Please update from main repository.

mitar commented on piplmesh/panels/map/templates/panel/map/panel.html:
```diff
@@ -3,22 +3,28 @@
-                'url': '{{ request.node.url|escapejs }}'
-            };
-            /* ]]> */
+{% addtoblock "css" %}<link href="{% static "piplmesh/panel/map/panel.css" %}" rel="stylesheet" type="text/css" media="screen" />{% endaddtoblock %}
```
Why you changed indentation? It was OK before. We indent on any Django template tag or XHTML block.

mitar commented on piplmesh/panels/map/templates/panel/map/panel.html:
```diff
@@ -3,22 +3,28 @@
+{% endaddtoblock %}
+<div id="basic-map" class="small">
+    <div id="basic-map-extend-img">
+        <img src="{% static "piplmesh/panel/map/images/extend.png" %}" width:40px height:90px> </img>
```
What is this `width:40px height:90px` I see there? Why `</img>`?

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -1,7 +1,44 @@
+var nodeLocation;
+var map;
+
+document.onkeydown = function(evt) {
```
O no no. Use jQuery for event handling.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -1,7 +1,44 @@
+var nodeName;
```
Why define global variables? Will they be accessed from some other JavaScript file? If not, it is probably no need for them.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -1,7 +1,44 @@
-    var nodeLocation = new google.maps.LatLng(node.latitude, node.longitude);
-    var myOptions = {
-        zoom: 15,
+    nodeName = $('<p/>').text(node.name).append(' | ');
```
You do know that in JavaScript you can define function inside of functions?

mitar replied:
So no need to pass values between functions through global variables. Simply define both variables and needed helper functions inside `ready`.

mitar replied:
[One example](https://github.com/wlanslovenija/mainpage/blob/master/mainpage/wlansi/static/wlansi/js/map.js).

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -1,7 +1,44 @@
+    $('#map-info').append(nodeName);
+    var timerMap;
+    define_map();
+    $("#basic-map-extend-img").click(function() {
```
We use `'` for constant strings, not `"`.

We write space before `()` in anonymous functions.

List all arguments of the function, even if you do not use them. Because in this way somebody after you will maybe get some idea what is possible to do more. In this case, `event` is passed.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -1,7 +1,44 @@
+    define_map();
+    $("#basic-map-extend-img").click(function() {
+        setExtendedMapControls();
+        open_advanced_map();
```
Why is one function written in camel case and other with underscores?

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -1,7 +1,44 @@
+    });
+});
+
+function getStyle(el,styleProp)
```
Space after comma.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -1,7 +1,44 @@
+
+function getStyle(el,styleProp)
+{
+    var x = document.getElementById(el);
```
Use jQuery.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -1,7 +1,44 @@
+});
+
+function getStyle(el,styleProp)
+{
```
Brace on the line before, at the end.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -1,7 +1,44 @@
+{
+    var x = document.getElementById(el);
+    if (x.currentStyle)
+        var y = x.currentStyle[styleProp];
```
Use braces around this line. It is more readable.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -1,7 +1,44 @@
+    nodeLocation = new google.maps.LatLng(node.latitude, node.longitude);
+    myOptions = {
+        zoom: 12,
         center: nodeLocation,
```
Always use quote key/label names as strings, not bare like this. So `'center'`.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -10,14 +47,56 @@ $(document).ready(function () {
-    var nodeWebsite = $('<a/>').prop('href', node.url).text(gettext("more info"));
-    nodeName.append(nodeWebsite);
-    $('#map_info').append(nodeName);
+    map = new google.maps.Map(document.getElementById("map"), myOptions);
```
jQuery, not `getElementById`.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -10,14 +47,56 @@ $(document).ready(function () {
+    map.set('streetViewControl', true);
+    map.setZoom(16);
+}
+function setBasicMapControls(){
```
Empty line in-between.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -10,14 +47,56 @@ $(document).ready(function () {
+    );
+}
+
+function setExtendedMapControls(){
```
The name is not really true. Because you also set zoom. Probably `configureExtendedMap` and `configureBasicMap` would be a better choice.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -10,14 +47,56 @@ $(document).ready(function () {
+    map.set('scaleControl', false);
+    map.set('draggable', false);
+    map.set('streetViewControl', false);
+    map.setZoom(12);
```
I don't like that you are duplicating basic map configuration here. Can you simply reset it to `myOptions`? (Which really have a bad name, please.)

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -10,14 +47,56 @@ $(document).ready(function () {
+
+function open_advanced_map(){
+    document.getElementById('basic-map-extend-img').style.visibility="hidden";
+    var div_overlay = jQuery('<div id="overlay"> </div>');
```
Use `$` instead of `jQuery`. No need for portability here.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -10,14 +47,56 @@ $(document).ready(function () {
+}
+
+function open_advanced_map(){
+    document.getElementById('basic-map-extend-img').style.visibility="hidden";
```
Use jQuery everywhere.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -10,14 +47,56 @@ $(document).ready(function () {
+    div_advanced_map.appendTo(document.body);
+    div_overlay.appendTo(document.body);
+    $('#advanced-map').append('<div id="advanced-map-container"></div>');
+    $('#advanced-map').append('<a class="close" onclick=close_advanced_map()></a>');
```
NO. Don't assign event handlers as this. And even don't create DOM elements as strings. This should be written as:

```
$('<a/>').addClass('close').click(close_advanced_map)
```

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -10,14 +47,56 @@ $(document).ready(function () {
+    var div_advanced_map = jQuery('<div id="advanced-map"></div>');
+    div_advanced_map.appendTo(document.body);
+    div_overlay.appendTo(document.body);
+    $('#advanced-map').append('<div id="advanced-map-container"></div>');
```
```
$('<div/>').attr('id', 'advanced-map-container')
```

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -10,14 +47,56 @@ $(document).ready(function () {
+    $('#map').detach().prependTo('#advanced-map-container');
+    $("#advanced-map").animate({
+        width: '86%'
+    }, 500, "linear", function(){
```
Spaces around `()`.

mitar commented:
Now also some general comments.

It will be possible to move panels around. So your idea that you open from the side is not good. You should pop-out the map, not slide it from side.

Map types available are not reset when you minimize it back. Initially, there is only Map dropbox, after close and minimization, Map and satellite are both displayed.

Animation is to slow. It should open faster.

I would have smaller icon for pop-out. Something in the corner, similar to what Google Talk has. Some an arrow pointing out of the corner.

Also map should look the same as when small, only bigger. Now zoom is changed and it looks really weird.

Maybe zoom of smaller map should be changed by default. Idea was, that it shows an area of 100-200 m wide.

Gitek1 committed:
Fixed:

Space before {.
Space after :.
Use jQuery for event handling.
Define local variables instead of global variables.
Define functions inside document ready function.
Use' for constant strings instead of ".
Define all function in camel case.
Use jQuery insted of document.getElementById().
Use quote key/label names as strings
Change some function's bad names.
Use $ instead of jQuery.
Replace DOM elements as strings with other method.
Add close button picture to static folder.

mitar commented:
Please comment when you do anything.

Please update from main repository.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.css:
```diff
@@ -1,4 +1,90 @@
+    margin-right: 20px;
+    position: relative;
 }
+#basic-map-extend-img {
```
Empty lines between CSS blocks.

mitar replied:
TODO.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.css:
```diff
@@ -1,4 +1,90 @@
+    height: 250px;
+    margin-bottom: 50px;
+    margin-right: 20px;
+    position: relative;
```
Why relative position here?

mitar replied:
TODO.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.css:
```diff
@@ -1,4 +1,90 @@
+    cursor: pointer;
+    height: 100px;
+    width: 49px;
+    z-index: 60000000;
```
You really don't have to exaggerate. Please use some smaller number. :-)

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.css:
```diff
@@ -1,4 +1,90 @@
+    border: none;
+}
+.close {
+    background-image: url(http://jquerytools.org/media/img/overlay/close.png);
```
Copy any external files to our static directory. But of course license should be checked. If you want, you can use the icon from jQuery Visage. There is also a close button available.

Anyway, I would design close button differently. So that it is not over the map in the corner, but in the corner of the window itself.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.css:
```diff
@@ -1,4 +1,90 @@
+    clear: none;
+    display: block;
+    background-color: black;
+    opacity:0.4;
```
Make it less transparent. Space before the value.

mitar commented on piplmesh/panels/map/templates/panel/map/panel.html:
```diff
@@ -3,22 +3,29 @@
-    <div id="map_info"></div>
-    <div id="map_canvas"></div>
-{% endblock %}
+{% addtoblock "css" %}<link href="{% static "piplmesh/panel/map/panel.css" %}" rel="stylesheet" type="text/css" media="screen" />{% endaddtoblock %}
```
Indent.

mitar commented on piplmesh/panels/map/templates/panel/map/panel.html:
```diff
@@ -3,22 +3,29 @@
+{% endaddtoblock %}
+<div id="basic-map" class="small">
+    <div id="basic-map-extend-img">
+        <img src="{% static "piplmesh/panel/map/images/extend.png" %}"> </img>
```
Why `</img>`?

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -1,23 +1,81 @@
-    };
-    var map = new google.maps.Map(document.getElementById('map_canvas'), myOptions);
+$(document).ready(function (event) {
+    var myOptions;
```
Please use our [code style](https://github.com/wlanslovenija/PiplMesh/pull/199#commitcomment-1707793).

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -1,23 +1,81 @@
+        var value = $(el).css(styleProp);
+        return value;
+    }
+    function defineMap(){
```
Ah, some empty lines would be so great. :-) And spaces after `()`.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -1,23 +1,81 @@
+        openAdvancedMap();
     });
-});
+    function getStyle(el, styleProp){
```
What is this function? Where it is used?

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -1,23 +1,81 @@
+            'mapTypeId': google.maps.MapTypeId.ROADMAP,
+            'streetViewControl': false
+        };
+        map = new google.maps.Map($("#map")[0], myOptions);
```
Static string.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -1,23 +1,81 @@
+        map.streetViewControl = true;
+    }
+    function configureBasicMap(){
+        map.set('navigationControl',false);
```
Decide. Do you use `.set` or not?

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -1,23 +1,81 @@
+    function closeAdvancedMap(){
+        $('#advanced-map').animate({
+            width: '-=86%'
+        }, 500, "linear", function (event){
```
Static string.

mitar replied:
Space after `(`.

mitar replied:
Ehm, `)`.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -1,23 +1,81 @@
+            width: '-=86%'
+        }, 500, "linear", function (event){
+            //callback
+            ($('#map').detach().prependTo('#basic-map')).insertAfter('#map-info');
```
Why you have `($('#map').detach().prependTo('#basic-map'))` and not `$('#map').detach().prependTo('#basic-map')`.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -1,23 +1,81 @@
+            ($('#map').detach().prependTo('#basic-map')).insertAfter('#map-info');
+            $('#overlay').remove();
+            $('#advanced-map').remove();
+            $('#basic-map-extend-img').css('visibility', 'visible');
```
`.show()`?

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -1,23 +1,81 @@
+        $('#map').detach().prependTo('#advanced-map-container');
+        $('#advanced-map').animate({
+            width: '86%'
+        }, 500, "linear", function (event){
```
Constant string.

mitar commented:
Use 'fast' speed and 'swing' easing. While resizing and the end, keep the marker in the center. Resize from all directions, not just from the side like you do now. (You can use jQuery UI effect for that.) Make map close when you click outside of the map. Make background less transparent. Make background fade in and not just appear.

mitar commented:
Please update from main repository.

Gitek1 committed:
Fixed:
 When we click button close of advanced map, the first element which becomes hidden is overlay ---> it looks better
 Copied external files to our static directory.
 Created div basic-map-extend-img instead img class basic-map-extend-img and shown img close button with JQuery.
 Used the icon close button from Visage and set it in the right top corner of the window instead of on the map in the corner, like it was before.
 Made layer background less transparent and fade in and fade out effect.
 Added function, which makes the map close when we click outside of the map.
 Fixed style of code: ($('#map').detach().prependTo('#basic-map')) to $('#map').detach().prependTo('#basic-map')
 Used method hide() instead of method css to set div visibility to false:  $('#basic-map-extend-img').css('visibility', 'hidden'); to >   $('#basic-map-extend-img').hide();
 Used method show() instead of method css to set div visibility to true:   $('#basic-map-extend-img').css('visibility', 'visible'); to $('#basic-map-extend-img').show();
 When we resize map, the marker stays still in center of the map.

To be continued...;)

Gitek1 committed:
My previous commit doesn't work properly because of one variable which had not be declared.
Simplified basic map configuration.
Created smaller icon for pop-out in top left corner.

Gitek1 committed:
Declared variable nodeLocation because i deleted it accidentally before the last commit.

Gitek1 committed:
Css fixed for close and extend map button.

mitar commented:
Please update from main repository. It cannot be merged automatically.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.css:
```diff
@@ -1,4 +1,96 @@
+    position: fixed;
+    top: 0px;
+    bottom: 0px;
+    left: 0px;
```
`0px` is redundant. `0` is better. It does not matter which unit it is, when it is 0.

Gitek1 replied:
Fixed.

Gitek1 committed:
Do resize map animation with fade in effect.

Gitek1 commented:
Fixed.

mitar commented:
Update from main repository.

Gitek1 committed:
Replaced all 0px with 0.

mitar commented on piplmesh/panels/map/templates/panel/map/panel.html:
```diff
@@ -3,22 +3,27 @@
+{% addtoblock "js" %}
+<script type="text/javascript">
+/* <![CDATA[ */
+var imageCloseUrl = "url("+"{% static "piplmesh/panel/map/images/close.png" %}"+")";
```
This is not the best way. You should keep global variables to minimum. This would be much better to define image in template, as HTML, just by default you do not display it (with CSS). And then you use it wherever you need.

mitar commented on piplmesh/panels/map/templates/panel/map/panel.html:
```diff
@@ -3,22 +3,27 @@
+/* <![CDATA[ */
+var imageCloseUrl = "url("+"{% static "piplmesh/panel/map/images/close.png" %}"+")";
+var imageExtendUrl = "url("+"{% static "piplmesh/panel/map/images/extend.png" %}"+")";
+var node = {
```
This should probably be moved from panel code to top PiplMesh code? Probably also other panels/code will need information about current node in JavaScript?

mitar commented on piplmesh/panels/map/templates/panel/map/panel.html:
```diff
@@ -3,22 +3,27 @@
+</script>
+{% endaddtoblock %}
+<div id="basic-map" class="small">
+    <div id="basic-map-extend-img"></div>
```
`img` -> `image`. Be verbose.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -1,23 +1,90 @@
+        google.maps.event.trigger(map, 'resize');
+        map.setCenter(center);
+    }
+    // TODO basic map resize animation should resize from its given position. That means if map is on the left side of the screen, animation should resize from the same side. Looks much prettier than a simple fade in pop up.
```
Write `:` after TODO string. Start with upper case letter. (Normal sentence.)

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -1,23 +1,90 @@
-        streetViewControl: false
-    };
-    var map = new google.maps.Map(document.getElementById('map_canvas'), myOptions);
+$(document).ready(function (event) {
```
He he. There is no event in this case. But yes, in general we want all arguments. But this one is useless. There is no real event behind and no data there.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -1,23 +1,90 @@
-    };
-    var map = new google.maps.Map(document.getElementById('map_canvas'), myOptions);
+$(document).ready(function (event) {
+    var nodeLocation;
```
Please respect our [code style](https://github.com/wlanslovenija/PiplMesh/pull/199#commitcomment-1707793). It might be that also previous code was not in this style, as it seems. Feel free to fix it.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -1,23 +1,90 @@
+            'title': node.name
+        });
+    }
+    function configureAdvancedMap(){
```
Space after `)`.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -1,23 +1,90 @@
+        map.scaleControl = true;
+        map.draggable = true;
+        map.streetViewControl = true;
+        map.set('mapTypeControlOptions',google.maps.ControlPosition.TOP_RIGHT);
```
Space after `,`.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -1,23 +1,90 @@
+    }
+    function closeAdvancedMap(){
+        $('#closeButton').remove();
+        $('#overlay').fadeOut('slow', 0.0, function() {
```
In anonymous functions, have spaces around `()`.

mitar commented on piplmesh/panels/map/static/piplmesh/panel/map/panel.js:
```diff
@@ -1,23 +1,90 @@
+    function openAdvancedMap(){
+        $('#basic-map-extend-img').hide();
+        $('<div/>').attr('id', 'advanced-map').appendTo(document.body);
+        $('<div/>').attr('id', 'overlay').fadeTo("slow", 0.8, function (){
```
Constant string.

mitar replied:
(Use only `'`.)

mitar commented:
Please go over all code and JavaScript code and clean the style.

Gitek1 committed:
Merge remote-tracking branch 'RemoteWlan/master' into map_panel

Fixed conflicts in:
	piplmesh/panels/map/static/piplmesh/panel/map/panel.css
	requirements.txt

Gitek1 committed:
1)I removed imageCloseUrl and imageExtendUrl from panel.html and instead of them added background-image in css. I transfered Node to home.html.
2) I renamed closeButton to close-button, so that it is in the style of other classes. I added him fadeTo when we resize basic map.
3) I deleted #map_canvas from css.
4) I paid attention to code style.

Gitek1 committed:
TODO for #basic-map in css.

Gitek1 commented:
Fixed.

mitar commented:
Please update from main repository to be able to merge.

mitar commented:
It is OK, but not incredible.

mitar commented:
I merged for you.

mitar commented:
(You can thank me next time.)
